### PR TITLE
Add case when expression for grouping sets when collations are used.

### LINKED ISSUE #11050: When the default collation is set to en_US, the result is incorrect.
### What happens?

```
set default_collation=c;
CREATE TABLE sales (
    product_id INT,
    region VARCHAR(50),
    year INT,
    amount_sold DECIMAL(10,2)
);

INSERT INTO sales VALUES 
(1, 'North', 2020, 1000.00),
(1, 'North', 2021, 1500.00),
(1, 'South', 2020, 800.00),
(1, 'South', 2021, 700.00),
(2, 'North', 2020, 500.00),
(2, 'North', 2021, 600.00),
(2, 'South', 2020, 400.00),
(2, 'South', 2021, 550.00);

SELECT product_id, region, SUM(amount_sold) AS total_amount
FROM sales
GROUP BY GROUPING SETS ((product_id), (region), ())
ORDER BY product_id, region, total_amount;

┌────────────┬─────────┬───────────────┐
│ product_id │ region  │ total_amount  │
│   int32    │ varchar │ decimal(38,2) │
├────────────┼─────────┼───────────────┤
│          1 │         │       4000.00 │
│          2 │         │       2050.00 │
│            │ North   │       3600.00 │
│            │ South   │       2450.00 │
│            │         │       6050.00 │
└────────────┴─────────┴───────────────┘
```
Change the default collation to a different language, such as en_us, results in incorrect (maybe).
```
set default_collation=en_us;
CREATE TABLE sales (
    product_id INT,
    region VARCHAR(50),
    year INT,
    amount_sold DECIMAL(10,2)
);

INSERT INTO sales VALUES 
(1, 'North', 2020, 1000.00),
(1, 'North', 2021, 1500.00),
(1, 'South', 2020, 800.00),
(1, 'South', 2021, 700.00),
(2, 'North', 2020, 500.00),
(2, 'North', 2021, 600.00),
(2, 'South', 2020, 400.00),
(2, 'South', 2021, 550.00);

SELECT product_id, region, SUM(amount_sold) AS total_amount
FROM sales
GROUP BY GROUPING SETS ((product_id), (region), ())
ORDER BY product_id, region, total_amount;

┌────────────┬─────────┬───────────────┐
│ product_id │ region  │ total_amount  │
│   int32    │ varchar │ decimal(38,2) │
├────────────┼─────────┼───────────────┤
│          1 │ North   │       4000.00 │
│          2 │ North   │       2050.00 │
│            │ North   │       3600.00 │
│            │ North   │       6050.00 │
│            │ South   │       2450.00 │
└────────────┴─────────┴───────────────┘
```

### To Reproduce

```
set default_collation=c;
CREATE TABLE sales (
    product_id INT,
    region VARCHAR(50),
    year INT,
    amount_sold DECIMAL(10,2)
);

INSERT INTO sales VALUES 
(1, 'North', 2020, 1000.00),
(1, 'North', 2021, 1500.00),
(1, 'South', 2020, 800.00),
(1, 'South', 2021, 700.00),
(2, 'North', 2020, 500.00),
(2, 'North', 2021, 600.00),
(2, 'South', 2020, 400.00),
(2, 'South', 2021, 550.00);

SELECT product_id, region, SUM(amount_sold) AS total_amount
FROM sales
GROUP BY GROUPING SETS ((product_id), (region), ())
ORDER BY product_id, region, total_amount;

┌────────────┬─────────┬───────────────┐
│ product_id │ region  │ total_amount  │
│   int32    │ varchar │ decimal(38,2) │
├────────────┼─────────┼───────────────┤
│          1 │         │       4000.00 │
│          2 │         │       2050.00 │
│            │ North   │       3600.00 │
│            │ South   │       2450.00 │
│            │         │       6050.00 │
└────────────┴─────────┴───────────────┘
```
Change the default collation to a different language, such as en_us, results in incorrect (maybe).
```
set default_collation=en_us;
CREATE TABLE sales (
    product_id INT,
    region VARCHAR(50),
    year INT,
    amount_sold DECIMAL(10,2)
);

INSERT INTO sales VALUES 
(1, 'North', 2020, 1000.00),
(1, 'North', 2021, 1500.00),
(1, 'South', 2020, 800.00),
(1, 'South', 2021, 700.00),
(2, 'North', 2020, 500.00),
(2, 'North', 2021, 600.00),
(2, 'South', 2020, 400.00),
(2, 'South', 2021, 550.00);

SELECT product_id, region, SUM(amount_sold) AS total_amount
FROM sales
GROUP BY GROUPING SETS ((product_id), (region), ())
ORDER BY product_id, region, total_amount;

┌────────────┬─────────┬───────────────┐
│ product_id │ region  │ total_amount  │
│   int32    │ varchar │ decimal(38,2) │
├────────────┼─────────┼───────────────┤
│          1 │ North   │       4000.00 │
│          2 │ North   │       2050.00 │
│            │ North   │       3600.00 │
│            │ North   │       6050.00 │
│            │ South   │       2450.00 │
└────────────┴─────────┴───────────────┘
```

### OS:

linux

### DuckDB Version:

0.10.0

### DuckDB Client:

cli

### Full Name:

Eric Wong

### Affiliation:

personal

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a nightly build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

### PR DESCRIPTION
**Pull request by Tmonster**
Fixes https://github.com/duckdb/duckdb/issues/11050

Grouping sets need to set all columns that are not used to NULL. When collations are introduced, there is a column count mismatch where the original varchar column is not set to null, but instead a `FIRST()` aggregate function is applied to it. The collated column is set to null however. This would cause incorrect results for grouping sets in the presence of collations, see the bug for an example.

The fix is to modify the expressions of the group by operator. Instead of just projecting the aggregate expression, we project a case statement that says `case when collated(col) is null then null else first(col)`

### CODE REVIEW

File: test/sql/collate/test_collate_and_grouping_sets.test [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+FROM sales
+GROUP BY GROUPING SETS ((product_id), (region), ())
+ORDER BY product_id, region, total_amount;
+
+query I
+select count(*) from (select * from t1 INTERSECT select * From t2);
+----
+5
+
+mode skip
```
Comment (Mytherin) [inline test/sql/collate/test_collate_and_grouping_sets.test:52]: Is this left in on purpose?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! LGTM - one minor comment. There also seems to be a CI failure in the force restart

File: test/sql/collate/test_collate_and_grouping_sets.test (hunk 1 above)
Comment (Tmonster) [reply test/sql/collate/test_collate_and_grouping_sets.test:52]: Ah, no, just forgot to take it out. Need to clean up this test too actually

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**